Feature/update pom

jweiser opened the pull request:
Reorganize and streamline the POM file. All details are in the commit messages.

jweiser committed:
adds IntelliJ files to .gitignore

jweiser committed:
reorganizes and updates POM file

- adds tags for packaging, description, URL, license, developer, scm
- updates name tag value
- reorganizes POM order to Maven recommended order (https://maven.apache.org/developers/conventions/code.html)
- updates release-common-lib to v1.2.0
- adds groupId and version tag for maven-assembly-plugin
- adds plugins to build section for compiling source, releasing new versions, and ensuring POM follows recommended order

jweiser committed:
reorganizes and adds comments to .gitignore file

jweiser committed:
updates runOrthoinference.sh bash script to use any jar file version

SolomonShorser-OICR commented on pom.xml:
```diff
@@ -1,16 +1,43 @@
+
 	<groupId>org.reactome.datapipeline</groupId>
 	<artifactId>orthoinference</artifactId>
 	<version>1.0.2-SNAPSHOT</version>
```
Still a snapshot version?

jweiser replied:
Yes, these updates aren't intended as a release yet.  Just updating some things for when the next release is made.

SolomonShorser-OICR commented on pom.xml:
```diff
@@ -55,21 +82,35 @@
+				<artifactId>maven-compiler-plugin</artifactId>
+				<version>3.8.1</version>
+				<configuration>
+					<source>${maven.compiler.source}</source>
```
I _think_ that if maven.compiler.source and ...target are defined as properties, maven is clever enough to use the properties and there is no need to define this plugin with these values. See: https://maven.apache.org/plugins/maven-compiler-plugin/examples/set-compiler-source-and-target.html

jweiser replied:
Okay, I've removed the property values and just have the values set in the compiler plugin itself so it the compiler plugin version and options are explicit

SolomonShorser-OICR replied:
Just curious why that and not remove the compiler plugin? If only the properties are left, the version is still explicit.

jweiser replied:
I could add the properties and remove the compiler plugin.  I was just thinking that the other projects also have the compiler plugin specified, it sets the version of the plugin instead of defaulting to the latest version to allow for stable/repeatable builds, and if we want to customize its configuration we can.

I don't have strong feelings about keeping the compiler plugin instead of the properties, though.  Do you think the properties are preferable?

I'd just need to go into the other projects POMs and swap out the compiler plugin for the maven compiler properties.

SolomonShorser-OICR replied:
No, I don't have any strong feelings, I was just surprised. I thought you would have gone the other way and simplified the POM by removing the plugin and just leaving the properties instead of leaving the compiler plugin with a specific version.

I don't think it matter strongly either way. Like I said, I was just surprised by which solution you chose. ;)

jweiser replied:
Yeah, I was considering both options.  I just thought it was nice (though a bit more verbose) to keep the compiler options in the build section with the other plugins and have the specific version of the compiler plugin nailed down.

SolomonShorser-OICR replied:
Hehe yeah I think I usually do it the other way, but both are fine. :)

jweiser committed:
removes property names for explicitly set compiler-plugin

https://maven.apache.org/plugins/maven-compiler-plugin/examples/set-compiler-source-and-target.html

cookersjs commented on .gitignore:
```diff
@@ -1,6 +1,13 @@
\ No newline at end of file
+logs/
+target/
+
```
I wonder if we should gitignore the default config.properties files in the release repos?

jweiser replied:
Yeah, it would definitely be useful to ignore changes to the default `config.properties` so nothing gets committed to it accidentally.

I was using this method: https://stackoverflow.com/questions/3319479/can-i-git-commit-a-file-and-ignore-its-content-changes to keep the default `config.properties` in the repo but use a shell script that runs the "assume unchanged" command run on the default and creates a new copy of the `config.properties` file that is what is used by the program and has a name that is in .gitignore.  That is, have git overlook changes to the original and .gitignore the working copy.

See this script: https://github.com/reactome/release-data-exporter/blob/develop/configureDataExporter.sh

cookersjs commented on pom.xml:
```diff
@@ -1,16 +1,40 @@
+	<packaging>jar</packaging>
+
+	<name>orthoinference</name>
+	<description>Orthology projection of human reactions and pathways to non-human species</description>
```
I think we should include that it is based on PANTHER in the description

jweiser replied:
Sure, I can add it in.

jweiser replied:
Done: https://github.com/reactome/release-orthoinference/pull/1/commits/86aaedd082034456644f459b77170bf560924194

cookersjs commented on runOrthoinference.sh:
```diff
@@ -9,11 +9,15 @@ git pull
 mvn clean compile assembly:single
 
+## Ensures the correct jar file is obtained regardless of orthoinference project version
+orthoinference_jar_file=$(ls target/orthoinference-*-jar-with-dependencies.jar)
```
As a thought, should this be the way Jenkinsfiles are organized? Right now jar versions are taken from Global config variables that need to be manually set. Maybe we should just use this wildcard way of doing things so that its less sticky for someone to run steps

jweiser replied:
If it can be done for the Jenkins files, it might be a good idea.  Using the wildcard frees us up from having to manually update the version.

The only thing is we need to be sure only one jar version exists at a time.  Adding it in the shell script is safe since the jar file name is found immediately after doing a `mvn clean` and rebuild ensuring an old `target` directory is removed and regenerated containing only the new jar version.

SolomonShorser-OICR replied:
> The only thing is we need to be sure only one jar version exists at a time.

A very good reason to _not_ do this. Just use a variable in the script to store the version number. Or if you really want, have maven update the script (there's probably a way to do this).

jweiser replied:
That's fair.  In the small shell script it's much easier to be sure the jar generated in the previous command is the only one.

In a larger system like Jenkins, it's doable but it might be easier to accidentally have additional jar files.

cookersjs replied:
> If it can be done for the Jenkins files, it might be a good idea. Using the wildcard frees us up from having to manually update the version.
> 
> The only thing is we need to be sure only one jar version exists at a time. Adding it in the shell script is safe since the jar file name is found immediately after doing a `mvn clean` and rebuild ensuring an old `target` directory is removed and regenerated containing only the new jar version.

This does also match how the Jenkinsfile does things though, each time a step is run it rebuilds the target folder

jweiser committed:
adds UTF-8 source encoding property

jweiser committed:
updates POM file description to mention homology data is from PANTHER

jweiser committed:
adds comments to maven build plugins

jweiser committed:
adds normal_event_skip_list.txt to .gitignore
